### ISSUE TITLE
code erreor

### ISSUE THREAD

**Issue by pipihou** [lethienhoa/Very-Deep-Convolutional-Networks-for-Natural-Language-Processing#3]
Hi, i've mentioned that the output Dim of Conv Block 64  is  1 x 72 x 64, how could you do convolution with the filter with shape 1*3*128*128? Dimensions not equal!

**Comment by pipihou** [comment]
I have to say your code is totally wrong!

**Comment by lethienhoa** [comment]
Hi pipihou, it's been quite a long I didn't revisit it on github. I've rewrittent it in Slim some months ago & i can get very close results to Conneau et al., (2016). I've updated it [here](https://github.com/lethienhoa/Very-Deep-Convolutional-Networks-for-Natural-Language-Processing/blob/master/model.py) ! thanks !

**Comment by dt-ahmed-touila** [comment]
Hi, i think the filters have the wrong shape that is why you have dimension error. This is a rectified version of the conv/ conv_blocks :
`     
 with tf.name_scope("first-conv-layer"):
            print(self.input_x.get_shape())
            filter_shape = [num_quantized_chars, cnn_filter_size, 1, num_filters_per_size[0]]
            W = tf.Variable(tf.truncated_normal(filter_shape, stddev=0.05), name="W")
            b = tf.Variable(tf.constant(0.1, shape=[num_filters_per_size[0]]), name="b")
            conv = tf.nn.conv2d(self.input_x, W, strides=[1, num_quantized_chars, 1, 1], padding="SAME", name="first-conv")
            h = tf.nn.bias_add(conv, b)
            print(h.get_shape())

        for i in range(0,4):
                with tf.name_scope("conv-block"):
                    for j in range(0,num_rep_block[i]):
                        with tf.name_scope("sub1"):
                            filter_shape = [1,cnn_filter_size,int(h.get_shape()[3]), num_filters_per_size[i]]
                            W1 = tf.Variable(tf.truncated_normal(filter_shape, stddev=0.05), name="W1")
                            conv1 = tf.nn.conv2d(h, W1, strides=[1, 1, 1, 1], padding="SAME", name="conv1")
                            # bias is replaced by batch_norm (same effect as beta in BN)
                            batch_norm1 = tf.contrib.layers.batch_norm(conv1,
                                                                          center=True, scale=True, decay=0.9,
                                                                          is_training=self.is_training )     
                            batch_norm1 = tf.identity(batch_norm1, name="bn1")

                            h1 = tf.nn.relu(batch_norm1, name="relu1")

                        with tf.name_scope("sub2"):
                            filter_shape2 = [1,cnn_filter_size,num_filters_per_size[i], num_filters_per_size[i]]
                            W2 = tf.Variable(tf.truncated_normal(filter_shape2, stddev=0.05), name="W2")
                            conv2 = tf.nn.conv2d(h1, W2, strides=[1, 1, 1, 1], padding="SAME", name="conv2")
                            batch_norm2 = tf.contrib.layers.batch_norm(conv2,
                                                                          center=True, scale=True, decay=0.9,
                                                                          is_training=self.is_training)  #, name='bn2')
                            batch_norm2 = tf.identity(batch_norm2, name="bn2")
                            h = tf.nn.relu(batch_norm2, name="relu2")

                if (i<>3):      
                    with tf.name_scope("max-pooling"):

                        h = tf.nn.max_pool(h, ksize=[1, 1, pooling_filter_size,1 ],
                                              strides=[1, 1, 2, 1], padding='SAME', name="pool")`

**Comment by pipihou** [comment]
Thanks for the updated code!
